sphinx/config.py

```diff
@@ -257,6 +257,9 @@ class Config(object):
         linkcheck_timeout = (None, None, [int]),
         linkcheck_workers = (5, None),
         linkcheck_anchors = (True, None),
+        # Anchors starting with /! are ignored since they are
+        # commonly used for dynamic pages
+        linkcheck_anchors_ignore = (["/#!"], None),
```

Comment:
That's true, though I haven't see to many URI's where foo#!bar indicated a dynamic resource, it always appeared to be only those where it matched `http://example.com/#!bar`.

I'll see if I can understand how to add a suitable test.